types.go

```diff
@@ -83,3 +61,70 @@ func (m PubKey) AsABCI() abci.PubKey {
 		Type: m.Type,
 	}
 }
+
+func (m ValidatorUpdates) Validate() error {
+	var err error
+	for _, v := range m.ValidatorUpdates {
+		err = errors.Append(err, v.Validate())
+	}
+	return err
+}
+
+// Deduplicate makes sure we only use the last validator update to any given validator.
+// For bookkeeping we have an option to drop validators with zero power, because those
+// are being remove by tendermint once propagated.
+func (m ValidatorUpdates) Deduplicate(dropZeroPower bool) []ValidatorUpdate {
+	duplicates := make(map[string]int)
+	cleanValidatorSlice := make([]ValidatorUpdate, 0, len(m.ValidatorUpdates))
+
+	for _, v := range m.ValidatorUpdates {
+		if dropZeroPower && v.Power == 0 {
```

Comment:
There is a slight bug here.

If I add a validator twice, once with power: 5, other with power: 0....

Running with dedup=false will show the validator with power: 0
Running with dedup=true will show the validator with power: 5 (turns second into noop)

This is a rare case but please fix the logic here